tests/test_stitch.py

```diff
@@ -0,0 +1,164 @@
+
+    region2 = Region(np.array([[1, 2], [3, 4]]), (slice(0, 2), slice(0, 2)))
+    region2[:, 1] = np.array([5, 6])
+    assert np.all(region2 == np.array([[1, 5], [3, 6]]))
+
+    # Test with slice only
+    subregion_slice_only = region1[0:2]
+    assert np.all(subregion_slice_only == np.array([[1, 2], [3, 4]]))
+    assert subregion_slice_only.key == (slice(0, 2, 1), slice(0, 2, None))
+
+    # Test with tuple of slices
+    subregion = region1[0:2, 0:1]
+    assert np.all(subregion == np.array([[1], [3]]))
+    assert subregion.key == (slice(0, 2, 1), slice(0, 1, 1))
+
+    # Test with Ellipsis
+    subregion_ellipsis = region1[...]
+    assert np.all(subregion_ellipsis == np.array([[1, 2], [3, 4]]))
+    assert subregion_ellipsis.key == (slice(0, 2, 1), slice(0, 2, 1))
+
+    # Test with Boolean indexing
+    bool_idx = np.array([[True, False], [False, True]])
+    subregion_bool = region1[bool_idx]
+    assert np.all(subregion_bool == np.array([1, 4]))
+
+    # Test with None indexing
+    subregion_none = region1[None, :]
+    assert np.all(subregion_none == np.array([[[1, 2], [3, 4]]]))
+
+
+@pytest.mark.parametrize("extra_shape", [tuple(), (3,)])
+def test_Frame(extra_shape):
+    shape = (10, 10) + extra_shape
+
+    data = np.empty(shape, dtype=int)
+    data[...] = np.arange(data.size).reshape(shape)
+
+    frame = Frame()
+    frame[5:15, 5:15] = data
+
+    # Ranges need to be explicit
+    # TODO: Actually, we can guess it from assigned data
+    with pytest.raises(IndexError, match="Can not infer stop from shape"):
+        frame[:, :] = data
+
+    # Shapes of assigned arrays have to match
+    with pytest.raises(ValueError, match="Mixed shape is not allowed"):
+        frame[5:15, 5:15] = data[..., None]
+
+    assert frame.ndim == len(shape)
+    assert frame.shape == (15, 15) + extra_shape
+    assert frame._shape == (None, None) + extra_shape
+
+    regions = list(frame.iter_regions())
+    assert len(regions) == 1
+
+    # Make sure that all start/stop of the key are explicit
+    assert regions[0].key == (slice(5, 15), slice(5, 15))
+
+    # Convert to array
+    frame_array = np.asarray(frame)
+    assert frame.shape == frame_array.shape
+    np.testing.assert_equal(frame_array[5:15, 5:15], data)
+
+    # Requested region is partially outside of stored regions
+    # Overlap to the right is truncated (like with np.ndarray)
+    # Overlap to the left is padded
+    region = frame[10:20, 0:15]
+    assert region is not None
+    assert region.key == (slice(10, 15), slice(0, 15))
+    assert region.shape == (5, 15) + extra_shape
+    np.testing.assert_equal(region, np.pad(data[5:, :], ((0, 0), (5, 0)) + ((0, 0),) * len(extra_shape)))  # type: ignore
+
+    float_region = region.astype(float)
+    assert float_region.key == region.key
+
+    assert isinstance(region.sum(axis=0), np.ndarray)
+
+    with pytest.raises(IndexError, match="Invalid key: .* Only slice and tuple of slice are allowed."):
```

Comment:
You shouldn't test `_validate_key` (which is private) but `__getitem__` and `__setitem__` which make use of it.